aerleon/lib/juniper.py

```diff
@@ -781,6 +782,19 @@ def _MinimizePrefixes(
 
         return include_result, exclude_result
 
+    def _ReplaceICMPv6(self):
+        """Juniper has deprecated the use ic icmpv6 in favor of icmp6.
+        https://github.com/aerleon/aerleon/issues/323
+        """
+        if 'icmpv6' in self.term.protocol:
+            i = self.term.protocol.index('icmpv6')
+            del self.term.protocol[i]
+            self.term.protocol.insert(i, 'icmp6')
```

Comment:
```suggestion
            self.term.protocol[i] = 'icmp6'
```

Reply:
Also the index variable is named `loc` elsewhere in `Term.__init__`, may as well name it that, as `i` is conventionally used for numeric loop counters.

Reply:
Done

Reply:
the change from i to loc is sane but why the suggestion of `self.term.protocol[i] = 'icmp6'`?
If the list is empty it will return an `IndexError` and if it is not we are overwriting a valid entry

```
>>> foo = ['a','b']
>>> loc = foo.index('a')
>>> del foo[loc]
>>> foo
['b']
>>> foo[loc] = 'a'
>>> foo
['a']
```

Reply:
The list cannot be empty here because you are guarding with `if 'icmpv6' in self.term.protocol:`.

Reply:
d'oh